### PR TITLE
updated the example of custom-routes-proxying to utilize the App Router

### PR DESCRIPTION
**Pull request by Sam-Phillemon9493**
This PR updates the custom-routes-proxying example to use the App Router. Here are the changes that have been made:

- Renamed the "pages" folder to the "app" folder.
- Updated the routing for /, /about & /hello/[slug] files to align with the App Router.
- Added the layout.tsx file as part of the App Router.
- Updated the package.json file.
- Updated the Readme file.

CC: @samcx

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 9701821ce48ea22a56f9a03f5f3a82b112d8e981

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: examples/custom-routes-proxying/app/hello/[slug]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import Link from "next/link";
+
+export default function About({ params }) {
```
Comment (leerob) [inline examples/custom-routes-proxying/app/hello/[slug]/page.tsx:3]: Let's make sure we run Prettier.

We'll also want to type `params` here.

File: examples/custom-routes-proxying/package.json [hunk 2]
```diff
@@ -6,14 +6,14 @@
     "start": "next start"
   },
   "dependencies": {
-    "next": "latest",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "next": "^14.2.5",
```
Comment (leerob) [inline examples/custom-routes-proxying/package.json:9]: pin to `latest`

File: examples/custom-routes-proxying/app/hello/[slug]/page.tsx (hunk 1 above)
Comment (Sam-Phillemon9493) [reply examples/custom-routes-proxying/app/hello/[slug]/page.tsx]: @leerob I have made the changes.

File: examples/custom-routes-proxying/package.json (hunk 2 above)
Comment (Sam-Phillemon9493) [reply examples/custom-routes-proxying/package.json]: @leerob I have made the changes.

**Review by samcx — APPROVED** [review]
<picture data-single-emoji=":lgtm:" title=":lgtm:">:lgtm:</picture>

**Pull request merged**